examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts

```diff
@@ -29,8 +29,18 @@ export class ExecuteVsmCycleCommandHandler
       await this.txStorageAgent.updateTransactionStatus(tx);
 
       if (!this.agent.validateTransactionForExecution(tx)) {
-        tx.updateStatusToInvalid();
-        await this.txStorageAgent.updateTransactionStatus(tx);
+        this.eventBus.publish(
+          new WorkstepExecutionFailuresEvent(
+            tx.id,
```

Comment:
I think you can pass the full tx as a param and then reference the required fields in the event handler from the tx object. This  way we reduce the number of params this ctor takes.